layer/clients/project_service.py

```diff
@@ -81,6 +92,38 @@ def get_project_by_id(self, project_id: UUID) -> Optional[Project]:
             raise generate_client_error_from_grpc_error(err, "internal")
         return None
 
+    def delete_account_projects(
+        self,
+        account_id: uuid.UUID,
+        query: DeleteProjectsQuery,
+        dry_run: bool = True,
+        ignore_errors: bool = True,
+    ) -> int:
+        """
+        Used for e2e test cleanup
+        """
+        resp: GetAccountProjectsByIdResponse = self._service.GetAccountProjectsById(
+            GetAccountProjectsByIdRequest(account_id=AccountId(value=str(account_id)))
+        )
+
+        def should_be_deleted(p: ProjectMessage) -> bool:
+            return (
+                p.created_time.ToDatetime() < query.created_before
+                and re.search(query.name_matches, p.name) is not None
```

Comment:
Nit: can we call the variable `project`? Makes it easier to read

Reply:
in such a small function there can be debate :) but since you ask, I will update it